config/config.go

```diff
@@ -299,24 +312,48 @@ func ParseConfigString(configString string, terragruntOptions *options.Terragrun
 // an include block)
 func decodeAsTerragruntInclude(file *hcl.File, filename string, terragruntOptions *options.TerragruntOptions) (*terragruntInclude, error) {
 	terragruntInclude := terragruntInclude{}
-	err := decodeHcl(file, filename, &terragruntInclude, terragruntOptions, nil)
+	err := decodeHcl(file, filename, &terragruntInclude, terragruntOptions, nil, nil)
 	if err != nil {
 		return nil, err
 	}
 	return &terragruntInclude, nil
 }
 
-func decodeAsTerragruntConfigFile(file *hcl.File, filename string, terragruntOptions *options.TerragruntOptions, include *IncludeConfig) (*terragruntConfigFile, error) {
+// This decodes only the `locals` attribute of a terragrunt config, so its value can be used while decoding the rest of
+// the config. For consistency, `locals` in the call to `decodeHcl` is always assumed to be nil.
+func decodeAsTerragruntLocals(file *hcl.File, filename string, terragruntOptions *options.TerragruntOptions) (*terragruntLocals, error) {
+	terragruntLocals := terragruntLocals{}
+	err := decodeHcl(file, filename, &terragruntLocals, terragruntOptions, nil, nil)
```

Comment:
What happens if a local references another local?

Reply:
Ah I didn't think about this possibility. I think currently it will panic or raise a syntax error, but it would be nice if it can be recursive. Let me do a quick spike to see how hard it is to implement this.